components/server/src/api/workspace-service-api.ts

```diff
@@ -174,4 +185,107 @@ export class WorkspaceServiceAPI implements ServiceImpl<typeof WorkspaceServiceI
         response.workspace = this.apiConverter.toWorkspace(info);
         return response;
     }
+
+    async stopWorkspace(req: StopWorkspaceRequest): Promise<StopWorkspaceResponse> {
+        if (!req.workspaceId) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "workspaceId is required");
+        }
+        await this.workspaceService.stopWorkspace(ctxUserId(), req.workspaceId, "stopped via API");
+        const response = new StopWorkspaceResponse();
+        return response;
+    }
+
+    async deleteWorkspace(req: DeleteWorkspaceRequest): Promise<DeleteWorkspaceResponse> {
+        if (!req.workspaceId) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "workspaceId is required");
+        }
+        await this.workspaceService.deleteWorkspace(ctxUserId(), req.workspaceId, "user");
+        const response = new DeleteWorkspaceResponse();
+        return response;
+    }
+
+    async updateWorkspace(request: UpdateWorkspaceRequest): Promise<UpdateWorkspaceResponse> {
+        if (!request.workspaceId) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "workspaceId is required");
+        }
+        const userId = ctxUserId();
+
+        // check if user can access workspace first
+        await this.workspaceService.getWorkspace(userId, request.workspaceId);
+
+        const tasks: Array<Promise<any>> = [];
+
+        if (request.name) {
```

Comment:
if there are not updates provided other API will throw invlalid argument, we should be consisten.